src/helpers/shifts.helper.ts

```diff
@@ -58,33 +62,33 @@ export class ShiftHelper {
       wn1 > wn2 ? 1 : wn1 < wn2 ? -1 : 0
     );
     sortedShifts.forEach(({ workerName, shifts }) => {
-      const category = workerTypes[workerName] || "";
+      const category = workerTypes[workerName] ?? "";
       grouped[category][workerName] = shifts;
     });
     return grouped;
   }
+  public static isWorkingShift(shift: ShiftCode) {
+    return this.shiftCodeToWorkTime(shift) === 0;
+  }
 
-  public static rowWorkHoursInfo(
-    dataRow: DataRow,
-    scheduleLogic: ScheduleLogic | null,
-    sectionKey
-  ): [number, number, number] {
-    if (!sectionKey) return [0, 0, 0];
-    const rowData = dataRow?.rowData(true, false) ?? [];
-    const monthLogic = scheduleLogic?.sections.Metadata?.monthLogic;
-    const workingNorm =
-      (monthLogic?.workingDaysNumber || 0) *
-      WORK_HOURS_PER_DAY *
-      (scheduleLogic?.getSection<ShiftsInfoLogic>(sectionKey)?.availableWorkersWorkTime[
-        dataRow.rowKey
-      ] || 1);
-    const numberOfPreviousMonthDays = monthLogic?.numberOfPreviousMonthDays;
-    const workingHours = rowData
-      .slice(numberOfPreviousMonthDays)
-      .reduce((previousValue, currentValue) => {
-        return previousValue + ShiftHelper.shiftCodeToWorkTime(currentValue);
-      }, 0);
-    return [workingNorm, workingHours, workingHours - workingNorm];
+  public static caclulateWorkHoursInfo(
+    shifts: ShiftCode[],
+    workerNorm: number,
+    dates: VerboseDate[],
+    currentMonth: string
+  ) {
+    if (shifts.length !== dates.length) {
+      throw Error("Shifts should be defined for each day");
+    }
+    const firstDayOfCurrentMonth = dates.findIndex((d) => d.month === currentMonth);
+    const monthData = ArrayHelper.zip(shifts, dates).slice(firstDayOfCurrentMonth);
+    const workingData = monthData.filter(
+      (d) => VerboseDateHelper.isWorkingDay(d[1]!) && !this.isNotWorkingShift(d[0]!)
+    );
+    const requiredHours = workerNorm * WORK_HOURS_PER_DAY * workingData.length;
+    const actualHours = workingData.reduce((a, s) => a + this.shiftCodeToWorkTime(s[0]!), 0);
+    const overtime = actualHours - requiredHours;
+    return [requiredHours, actualHours, overtime];
```

Comment:
@pectom , @prenc   zerknijcie proszę na to. 
Logika jest taka, że najpierw szukamy pierwszy dzień miesiąca, któryjest potrzebny żeby wybrać aktualne zmiany, potem szukamy w tych zmianach wszystkie dni w których da się pracować ( nie święto, nie weekend, nie U,  nie L4 ). 
Na podstawie tych dni oraz etatu, liczymy normę. 
![image](https://user-images.githubusercontent.com/21079319/100533515-a2a6d580-3205-11eb-8abb-22b06d0574e6.png)